From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 13 Aug 2016 14:47:00 +0200
Last-Updated: Sun, 20 May 2018 16:28:00 +0200
Subject: [PATCH] Fix test build-auth:http_auth_offered

The test fails because of hard-coded libgit2 version, that may not be
the same as actually used libgit2...

--- a/src/tools/cargo/tests/testsuite/build_auth.rs
+++ b/src/tools/cargo/tests/testsuite/build_auth.rs
@@ -17,7 +17,7 @@
     let addr = server.local_addr().unwrap();
 
     fn headers(rdr: &mut BufRead) -> HashSet<String> {
-        let valid = ["GET", "Authorization", "Accept", "User-Agent"];
+        let valid = ["GET", "Authorization", "Accept"];
         rdr.lines()
             .map(|s| s.unwrap())
             .take_while(|s| s.len() > 2)
@@ -29,7 +29,6 @@
     let t = thread::spawn(move || {
         let mut conn = BufStream::new(server.accept().unwrap().0);
         let req = headers(&mut conn);
-        let user_agent = "User-Agent: git/2.0 (libgit2 0.27.0)";
         conn.write_all(
             b"\
             HTTP/1.1 401 Unauthorized\r\n\
@@ -42,7 +41,6 @@
             vec![
                 "GET /foo/bar/info/refs?service=git-upload-pack HTTP/1.1",
                 "Accept: */*",
-                user_agent,
             ].into_iter()
                 .map(|s| s.to_string())
                 .collect()
@@ -64,7 +62,6 @@
                 "GET /foo/bar/info/refs?service=git-upload-pack HTTP/1.1",
                 "Authorization: Basic Zm9vOmJhcg==",
                 "Accept: */*",
-                user_agent,
             ].into_iter()
                 .map(|s| s.to_string())
                 .collect()
